tests/ClassFixturesHelper.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types=1);
+
+use Symfony\Component\Yaml\Yaml;
+use function PHPUnit\Framework\assertArrayHasKey;
+use function PHPUnit\Framework\assertEquals;
+
+/**
+ * This class contains helpers for testing expected class structure.
+ */
+class ClassFixturesHelper
+{
+    private array $fixtures = [];
+
+    public function __construct()
+    {
+        foreach (glob(__DIR__ . '/fixtures/*.yaml') as $file) {
+            $this->fixtures += Yaml::parse(file_get_contents($file));
+        }
+    }
+
+    /**
+     * Voor elke method in de fixture wordt het volgende bevestigd:
+     * - De class bevat alle gespecificeerde properties met het juiste type
+     * - De class bevat een method met de juiste naam
+     * - Elke parameter in de fixture bestaat op de method
+     * - Elke method parameter heeft het juiste type
+     */
+    public function assertMatches(string $path, string $class): void
+    {
+        $shortName = (new ReflectionClass($class))->getShortName();
+        $fixture = $this->fixtures[$shortName];
+        $reflection = $this->reflect($path, $class);
+
+        foreach ($fixture['properties'] as $name => $type) {
+            assertArrayHasKey($name, $reflection['properties'], "De class moet de property `$name` bevatten.");
+            assertEquals($type, $reflection['properties'][$name], "De class property `$name` moet van het type `$type` zijn.");
+        }
+
+        foreach ($fixture['methods'] as $method => $params) {
+            assertArrayHasKey($method, $reflection['methods'], "De class moet de method `$method` bevatten.");
+
+            $reflectedMethod = $reflection['methods'][$method];
+            foreach ($params as $name => $type) {
+                assertArrayHasKey($name, $reflectedMethod, "De methode `$method` moet de parameter `$name` bevatten.");
+                assertEquals($type, $reflectedMethod[$name], "De parameter `$name` op de methode `$method` moet van het type `$type` zijn.");
+            }
+        }
+    }
+
+    /**
+     * Reflect class and parse it into the same structure as the yaml.
+     */
+    private function reflect(string $path, string $class)
+    {
+        require_once $path;
+        $reflection = new ReflectionClass($class);
+
+        $properties = $methods = [];
+        foreach ($reflection->getProperties() as $property) {
+            $properties[$property->name] = $property->getType()->getName();
+        }
+        foreach ($reflection->getMethods() as $method) {
+            $params = [];
+            foreach ($method->getParameters() as $param) {
+                $params[$param->name] = $param->getType()->getName();
+            }
+            $methods[$method->name] = $params;
+        }
+
+        return compact('properties', 'methods');
+    }

```

Comment:
Omdat ik het niet kon laten om de foreach loop te herschrijven naar een map, en er toen achter kwam dat `array_map` niet kon wat ik wou en het maar op mijn eigen manier ging doen.

```suggestion
        $properties = ClassFixturesHelper::getMappedProperties($reflection);
        $methods = ClassFixturesHelper::getMappedMethods($reflection);

        return compact('properties', 'methods');
    }

    static private function getMappedProperties(ReflectionClass $reflection): array
    {
        return array_reduce(
            $reflection->getProperties(),
            function (array $result ,ReflectionProperty $property) {
                return [...$result, $property->name => $property->getType()->getName()];
            }, []
        );
    }

    static private function getMappedParameters(ReflectionMethod $reflection): array
    {
        return array_reduce(
            $reflection->getParameters(),
            function (array $result ,ReflectionParameter $param) {
                return [...$result, $param->name => $param->getType()->getName()];
            }, []
        );
    }

    static private function getMappedMethods(ReflectionClass $reflection): array
    {
        return array_reduce(
            $reflection->getMethods(),
            function (array $result ,ReflectionMethod $method) {
                return [...$result, $method->name => ClassFixturesHelper::getMappedParameters($method)];
            }, []
        );
    }
```

Reply:
Ik vind ook dat dit roept om een `map`, maar die foreaches zijn wel lekker simpel en compact he? Ik zou deze PR eerlijk  gezegd op de werkvloer normaal gesproken niet accepteren, omdat ik bang ben dat de meeste php developers dit moeilijker te begrijpen vinden.
En het is ongebruikelijk om de methods static te maken. Waarom kies je daarvoor?

Reply:
Allemaal valide punten!  
Ik denk ook dat het met foreach loops beter leesbaar is, maar had al teveel moeite gestopt in een werkend alternatief vinden.  
Ik wou eerst een arrow function maken en die als static property opslaan om in de main method aan een reduce funcite te geven. Dat kan helaas niet, dus dit was de uitkomst, de static is soort van blijven staan omdat het kon.  
Ook voor naamgeving vind ik me helemaal in jou punt.  

Tnx voor de comment, ik was hiermee aan het spelen in de trein en wou een plekje om het te delen. Denk dat het geen daadwerkelijke improvement is.

Reply:
Top, dan sluit ik hem. Laravel's `Collection->mapWithKeys` was hier waarschijnlijk de schoonste oplossing, maar het is wel goed :)

Reply:
Die `MapWithKeys` is inderdaad precies wat ik wilde gebruiken, maar om daarvoor nou de Collections dependency te installeren... Toch maar hopen dat PHP 8.2 ingebouwde collections heeft :)